docs/proposals/toXml_fromXml_apis.md

```diff
@@ -0,0 +1,129 @@
+Therefore, It would be easy for them if we provided APIs to convert.
+
+Note: This feature(fromRecord()) is also required by the connector team. When they write a connector for the SOAP backend service(e.g Netsuite Connector),
+they need to convert the Ballerina record value to XML payload. As mentioned in the summary, we don't have a way to do this conversion. 
+They are planning to remove their custom implementation and use `xmldata` module, once we have this feature. As per the [discussion](https://github.com/ballerina-platform/ballerina-standard-library/issues/2819), 
+we are planning to introduce `fromXml` API for their requirement which is an improved API from `fromRecord`. 
+This will be used to do the conversion of the Ballerina record|map data to XML.
+
+## Description
+
+### APIs definition:
+
+```ballerina
+# Converts an XML to its `Map` or `Record` representation.
+# The namespaces and attributes will not be considered a special case.
+#
+# + xmlValue - The XML source to be converted to a given target type
+# + returnType - The `typedesc` of the `map<anydata>` that should be returned as a result
+# + return - The given target type representation of the given XML on success,
+# else returns an `xmldata:Error`
+public isolated function fromXml(xml xmlValue, typedesc<(map<anydata>)> returnType = <>) returns returnType|Error;
+```
+
+```ballerina
+# Converts a `Map` or `Record` representation to its XML representation.
+# The record has annotations to configure namespaces and attributes,  but others don't have these.
+#
+# + mapValue - The `Map` or `Record` representation source to be converted to XML
+# + return - XML representation of the given source if the source is
+# successfully converted or else an `xmldata:Error`
+public isolated function toXml(map<anydata> mapValue) returns xml|Error;
+```
+
+### Record Annotation Definitions:
+
+```ballerina
+# Defines the new name of the name.
+#
+# + value - The value of the new name
+public type NameConfig record {|
+    string value;
+|};
+```
+
+```ballerina
+# The annotation is used to specify the new name of the existing record name or field name according to the XML format.
+public annotation NameConfig Name on type, record field;
+# Defines the namespace of the XML element
+#
+# + prefix - The value of the prefix of the namespace
+# + uri - The value of the URI of the namespace
+public type NamespaceConfig record {|
+    string prefix;
+    string uri?;
+|};
+
+# The annotation is used to specify the namespace's prefix and URI of the XML element.
+public annotation NamespaceConfig Namespace on type, record field;
+```
+
+```ballerina
+# The annotation is used to denote the field that is considered an attribute.
+public annotation Attribute on record field;
+```
+### Rules for performing conversions between Map record and XML
+
+The same rules of conversion between JSON and XML are used. But, this doesn't consider the attributes and namespaces a special case.
```

Comment:
No. we need to specifically mention, that we are following OpenAPI specifications. Please reword this. provide reference link

Reply:
Sorry, I have to remove the `record` word in the heading. 

This section has the rules of conversion between XML and record.
https://github.com/ballerina-platform/module-ballerina-xmldata/pull/432/files#diff-0af47c0e4375c73ef387e816b3b5270631c76efdfef7a3cb92288fa0d919f176R86